packages/utxo-global/src/ckb/index.ts

```diff
@@ -0,0 +1,136 @@
+   */
+  get signType(): ccc.SignerSignType {
+    return ccc.SignerSignType.CkbSecp256k1;
+  }
+
+  constructor(
+    client: ccc.Client,
+    public readonly provider: Provider,
+  ) {
+    super(client);
+  }
+
+  getInternalAddress(): Promise<string> {
+    return this.getAccount();
+  }
+
+  async getIdentity(): Promise<string> {
+    return await this.getPublicKey();
+  }
+
+  async getAddressObj(): Promise<ccc.Address | undefined> {
+    const address = await this.getInternalAddress();
+    return await ccc.Address.fromString(address, this.client);
+  }
+
+  async getAddressObjs(): Promise<ccc.Address[]> {
+    const address = await this.getAddressObj();
+
+    if (!!address) {
+      return [address];
+    }
+
+    throw new Error("address not found");
+  }
+
+  async getAccount() {
+    const accounts = await this.provider.getAccount();
+    return accounts[0];
+  }
+
+  async getPublicKey(): Promise<ccc.Hex> {
+    const pubKeys = await this.provider.getPublicKey();
+    const account = await this.getAccount();
+    const pubKey = pubKeys.find((p) => p.address === account);
+
+    if (!pubKey) {
+      throw new Error("pubKey not found");
+    }
+
+    return ccc.hexFrom(pubKey.publicKey);
+  }
+
+  async connect(): Promise<void> {
+    await this.provider.connect();
+  }
+
+  async isConnected(): Promise<boolean> {
+    return await this.provider.isConnected();
+  }
+
+  async signMessageRaw(message: string | ccc.BytesLike): Promise<string> {
+    const challenge =
+      typeof message === "string" ? message : ccc.hexFrom(message).slice(2);
+    const account = await this.getAccount();
+    return this.provider.signMessage(challenge, account);
+  }
+
+  async signMessage(message: ccc.BytesLike): Promise<ccc.Signature> {
+    return {
+      signature: await this.signMessageRaw(message),
+      identity: await this.getIdentity(),
+      signType: ccc.SignerSignType.CkbSecp256k1,
+    };
+  }
+
+  async verifyMessage(
+    message: string | ccc.BytesLike,
+    signature: string | ccc.Signature,
+  ): Promise<boolean> {
+    if (typeof signature === "string") {
+      return this.verifyMessageRaw(message, signature);
+    }
+
+    if (
+      signature.identity !== (await this.getIdentity()) ||
+      ![this.signType, ccc.SignerSignType.CkbSecp256k1].includes(
+        signature.signType,
+      )
+    ) {
+      return false;
+    }
+
+    return this.verifyMessageRaw(message, signature.signature);
+  }
+
+  async verifyMessageRaw(
+    message: ccc.BytesLike,
+    signature: string | ccc.Signature,
+  ): Promise<boolean> {
+    const pubKey = await this.getPublicKey();
+    return ccc.verifyMessageCkbSecp256k1(message, signature as string, pubKey);
+  }
+
+  async signOnlyTransaction(
+    txLike: ccc.TransactionLike,
+  ): Promise<ccc.Transaction> {
+    const rawTx = cccA.JsonRpcTransformers.transactionFrom(txLike);
+    const txSigned = await this.provider.signTransaction(rawTx);
+    return JSON.parse(txSigned) as ccc.Transaction;
```

Comment:
The ccc.Transaction has extra props like methods, use ccc.Transaction.from instead.